File: ServerAndAngular/ORM/Project.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SoupDiscover.ORM
{
    /// <summary>
    /// A repository to process or processed
    /// </summary>
    public class Project
    {
        /// <summary>
        /// The primary  key of the project
        /// </summary>
        [Key]
        public int Id { get; set; }

        /// <summary>
        /// The repository where search files
        /// </summary>
        public virtual Repository Repository { get; set; }

        /// <summary>
        /// The forign key to the repository definition
        /// </summary>
        public int RepositoryId { get; set; }

        /// <summary>
        /// All packages found for this repository
        /// </summary>
        public virtual ICollection<Package> Packages { get; set; }

        /// <summary>
        /// The current status of the project
        /// </summary>
        public ProcessStatus ProcessStatus { get; set; }
    }
}

File: ServerAndAngular/ClientApp/src/app/model/Project.ts
import { RepositoryDto } from "./repository";
import { PackageDto, PackageType } from "./package";

export class ProjectDto {
  /**
   * The primary  key of the project
   * */
  name: string;

  /**
   * The repository where search files
   * */
  commandLinesBeforeParse: string;

  /**
   * The foreign key to the repository definition
   */
  repositoryId: string;

  /**
   * List of Url nuget server where find metadata of packages
   */
  nugetServerUrl: string;

  /**
   * The last analysis error
   */
  lastAnalysisError?: string;

  /**
   * The status of executing
   */
  processStatus?: ProcessStatus;

  /**
   * Date of the last analysis
   */
  lastAnalysisDate?: Date;
}

export enum ProcessStatus {
  /**
   * No started
   */
  Waiting,

  /**
   * The last processing return an error
   * */
  Error,

  /**
   * Processing
   * */
  Running,
}
